share/availability/options.go

```diff
@@ -0,0 +1,344 @@
+package availability
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+var (
+	// ErrInvalidAvailOption is an error that is returned by FullAvailabilityPameters.Validate
+	// LightAvailabilityPameters.Validate and ShareParameters.Validate when supplied with invalid values.
+	// This error will also be returned by NewShareAvailability if supplied with an invalid option
+	ErrInvalidAvailOption = fmt.Errorf("availability: invalid option")
+
+	// ErrInvalidAvailOption is an error that is returned by DiscoveryParameters.Validate
+	// and ShareParameters.Validate when supplied with invalid values.
+	// This error will also be returned by NewDiscovery if supplied with an invalid option
+	ErrInvalidDiscOption = fmt.Errorf("discovery: invalid option")
+)
+
+// ErrInvalidOption is a utility function to dedup code for error-returning
+// when dealing with invalid parameter values.
+func ErrInvalidOption(err error, optionName string, value string) error {
+	return fmt.Errorf("%w: value %s cannot be %s", err, optionName, value)
+}
```

Comment:
Lets write what was the provided value, so user can relate to existing config and be sure it was parsed correctly. Also nit, but usually preferred direct wording how to fix the issue instead of reversed one. So instead of "value cannot be", better to use "value should be".